npa.js

```diff
@@ -50,11 +49,16 @@ function npa (arg) {
   if (nameparse && validName(nameparse[3]) &&
       (!nameparse[2] || validName(nameparse[2]))) {
     res.name = (nameparse[1] || "") + nameparse[3]
-    if (nameparse[2])
+    if (nameparse[2]) {
       res.scope = "@" + nameparse[2]
+      res.escapedName = escapeName(res.name)
```

Comment:
@othiym23 This is based on your comment in Slack:

> well, we need to be sure we’re only applying in scoped contexts

So I only applied escaping when a `scope` applies. But I can change this if you don't think it hurts anything. I could easily apply escaping whenever `name` is set.

I could even take it a step further and always apply escaping before returning, either similar to what I described in option 2 of OP, or maybe with a little `try .. finally` action.